src/decoratedProviders/decoratedProviderInterface.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2022 Byndyusoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface IDecoratedProvider {
```

Comment:
why file name not equal exported class? 
2 different conventions are used...

Reply:
For decorators: it is NestJS style, see [example](https://github.com/nestjs/nest/blob/master/packages/common/decorators/core/controller.decorator.ts)

For interfaces: We discussed what we will write `Interface` or `Type` at the end of the file